src/main/java/aquality/selenium/core/elements/DesiredState.java

```diff
@@ -0,0 +1,47 @@
+package aquality.selenium.core.elements;
+
+import org.openqa.selenium.WebElement;
+
+import java.util.function.Predicate;
+
+/**
+ * Defines desired state for element with ability to handle exceptions.
+ */
+public class DesiredState {
```

Comment:
can we make this class package-private? and do not we need to extract an interface of it?

Reply:
no, we cannot make it private, we will need it in child libraries.
no, I don't think that we need to extract an interface of it. This class is a simple model, and we expect it to remain the same